soql-environment/src/main/scala/com/socrata/soql/environment/Source.scala

```diff
@@ -0,0 +1,58 @@
+package com.socrata.soql.environment
+
+import scala.util.parsing.input.{Position, NoPosition}
+
+import com.rojoma.json.v3.codec.{JsonEncode, JsonDecode}
+import com.rojoma.json.v3.util.{SimpleHierarchyEncodeBuilder, SimpleHierarchyDecodeBuilder, InternalTag, AutomaticJsonEncodeBuilder, AutomaticJsonDecodeBuilder}
+
+import com.socrata.soql.jsonutils.HierarchyImplicits._
+import com.socrata.soql.jsonutils.PositionCodec
+
+sealed abstract class Source[+RNS] {
+  val scopedResourceName: Option[ScopedResourceName[RNS]]
+  val position: Position
+  def withPosition(pos: Position): Source[RNS]
+  def mapSRN[RNS2](f: ScopedResourceName[RNS] => ScopedResourceName[RNS2]): Source[RNS2]
+}
+
+object Source {
+  case class Anonymous(position: Position) extends Source[Nothing] {
```

Comment:
Can you put a short scala doc comment in here to help clarify how these `Source` types differ and how they're intended to be used?